### PR TITLE
refactor(CLI): use commander instead of oclif

### PR DESCRIPTION
**Pull request by prokopsimek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [x] Update readme.

### CODE REVIEW

**Comment by prokopsimek** [comment]
# [DX Scanner](https://dxscanner.io) Report <a href="https://dxscanner.io" target="_blank">DXScanner.io</a>
## https://github.com/DXHeroes/dx-scanner 

<h3>⚠️ Improvements with medium impact
</h3>
<details>
<summary>Break down large pull requests into smaller ones</summary>

<b>Large pull request are hard to code review and it reduces the probability of finding bugs. Split your PRs into logical units. Do not have PR with more than 1000 changes.</b> <a href="https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067">https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067</a>
</details>
<details>
<summary>Solve Issues Continuously</summary>

<b>Do not have an open Issues more than 60 days. Solve Issues continuously.</b> <a href="https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612">https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612</a>
</details>
<details>
<summary>Update Dependencies of Minor and Patch Level</summary>

<b>Keep the dependencies updated to eliminate security concerns and compatibility issues. Use, for example, npm-check-updates.</b> <a href="https://github.com/tjunnone/npm-check-updates">https://github.com/tjunnone/npm-check-updates</a>


 Name                             | New     | Current  
---------------------------------- | --------- | ----------
 @octokit/rest                    | 16.40.1 | 16.37.0  
 @types/eslint                    | 6.1.7   | 6.1.3    
 @types/js-yaml                   | 3.12.2  | 3.12.1   
 axios                            | 0.19.2  | 0.19.1   
 @types/jest                      | 25.1.1  | 24.9.0   
 codecov                          | 3.6.2   | 3.6.1    
 rimraf                           | 3.0.1   | 3.0.0    
 @octokit/rest                    | 16.40.1 | 16.37.0  
 simple-git                       | 1.131.0 | 1.130.0  
 @types/node                      | 13.5.2  | 12.12.25 
 @typescript-eslint/eslint-plugin | 2.18.0  | 2.16.0   
 @typescript-eslint/parser        | 2.18.0  | 2.16.0   
 ts-jest                          | 25.1.0  | 25.0.0   

  <b>Impact changed from high to medium.</b>
</details>
<details>
<summary>ESLint Without Errors</summary>

<b>Use the ESLint correctly. You have some errors.</b> <a href="https://eslint.org/">https://eslint.org/</a>
</details>
<h3>🔔 Improvements with minor impact
</h3>
<details>
<summary>Write Commit Messages by Convention</summary>

<b></b> <a href="https://www.conventionalcommits.org/">https://www.conventionalcommits.org/</a>


 Commit Message                                                       | Problems                                                               
---------------------------------------------------------------------- | ------------------------------------------------------------------------
 feat(core): CLI changed to multi-command                             | subject must not be sentence-case, start-case, pascal-case, upper-case 
                                                                      |                                                                        
 - small cleanup                                                      |                                                                        
                                                                      |                                                                        
 BREAKING CHANGE: now exists commands such as init, practices and run |                                                                        
 Add fixFlag test                                                     | subject may not be empty; type may not be empty                        
 Fix fixPattern regex                                                 | subject may not be empty; type may not be empty                        

</details>
<details>
<summary>Update Dependencies of Major Level</summary>

<b>Keep the dependencies updated to have all possible features. Use, for example, npm-check-updates.</b> <a href="https://github.com/tjunnone/npm-check-updates">https://github.com/tjunnone/npm-check-updates</a>


 Name                        | New    | Current  
----------------------------- | -------- | ----------
 @semantic-release/changelog | 5.0.0  | 3.0.6    
 @semantic-release/git       | 9.0.0  | 8.0.0    
 @types/jest                 | 25.1.1 | 24.9.0   
 @types/node                 | 13.5.2 | 12.12.25 
 @types/semver               | 7.1.0  | 6.2.0    
 semantic-release            | 17.0.1 | 16.0.2   

</details>
<details>
<summary>Use a JS Logging Library</summary>

<b>Use a logging library to avoid errors and even cyber attacks. The most widely used logging library in the JavaScript community is Winston.</b> <a href="https://www.npmjs.com/package/winston/">https://www.npmjs.com/package/winston/</a>
</details>


---



Implementation is not adoption.
We can help you with both. :-)
[dxheroes.io](https://dxheroes.io)



##### Found a bug? Please <a href="https://github.com/DXHeroes/dx-scanner/issues" target="_blank">report</a>.

File: src/index.ts [hunk 1]
```diff
@@ -1,129 +1,102 @@
...
-        fail,
-        recursive: flags.recursive,
-        ci: flags.ci,
-        fix: flags.fix,
-        fixPattern: flags.fixPattern,
-      });
-      const scanner = container.get(Scanner);
+    // error on unknown commands
+    cmder.on('*', () => {
+      // const notifier = updateNotifier({ pkg: 'dx-scanner' });
```
Comment (adelkahomolova) [inline src/index.ts]: finish it

File: src/index.ts [hunk 2]
```diff
@@ -1,129 +1,102 @@
...
 
-      scanResult = await scanner.scan({ determineRemote: false });
-    }
-    cli.action.stop();
-    notifier.notify({ isGlobal: true });
-
-    const hrend = process.hrtime(hrstart);
+      console.log('ahojjjj');
```
Comment (adelkahomolova) [inline src/index.ts]: ?

File: src/model.ts [hunk 3]
```diff
@@ -62,6 +62,17 @@ export enum ProgrammingLanguage {
...
+  json: boolean;
+  details: boolean;
+  recursive: boolean;
+  ci: boolean;
+  fail: PracticeImpact;
+  fix: boolean;
+  fixPattern: string;
+};
+
 // Old model starts here
```
Comment (adelkahomolova) [inline src/model.ts]: remove?

File: src/reporters/CLIReporter.ts [hunk 4]
```diff
@@ -77,6 +85,7 @@ export class CLIReporter implements IReporter {
     lines.push(italic(blue('We can help you with both. :-)')));
     lines.push(italic(blue('- https://dxheroes.io')));
     lines.push(reset(' '));
+    lines.push(grey(`You can run the command with option ${italic('-d')} or ${italic('--details')} to show detailed informations.`));
```
Comment (adelkahomolova) [inline src/reporters/CLIReporter.ts]: Does it show also if the user already used the flag?

File: src/index.ts [hunk 5]
```diff
@@ -1,129 +1,102 @@
...
+import _ from 'lodash';
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const pjson = require('../package.json');
+
+class DXScannerCommand {
+  static async run() {
+    const cmder = new commander.Command();
+    cmder.version(pjson.version);
+
+    // cmd: init
```
[the remark this answers is not in the document]
Comment (prokopsimek) [reply src/index.ts]: changed

File: src/index.ts (hunk 1 above)
Comment (prokopsimek) [reply src/index.ts]: wip

File: src/index.ts (hunk 2 above)
Comment (prokopsimek) [reply src/index.ts]: wip

File: src/model.ts (hunk 3 above)
Comment (prokopsimek) [reply src/model.ts]: no

File: src/reporters/CLIReporter.ts (hunk 4 above)
Comment (prokopsimek) [reply src/reporters/CLIReporter.ts]: good point! fixed

**Comment by prokopsimek** [comment]
@adelkahomolova ready for final review.

File: src/index.ts [hunk 6]
```diff
@@ -1,129 +1,102 @@
...
+    cmder
+      .command('run [path]')
+      .description('Scan your project for possible DX recommendations')
+      .option(
+        '-a --authorization <authorization>',
+        'credentials to the repository (in format "token" or "username:token"; can be set as ENV variable DX_GIT_SERVICE_TOKEN)',
+        process.env.DX_GIT_SERVICE_TOKEN || process.env.GITHUB_TOKEN,
+      )
+      .option('--ci', 'CI mode', process.env.CI === 'true')
+      .option('-d --details', 'print details in reports')
```
Comment (adelkahomolova) [inline src/index.ts]: Does it work without a comma between -d and --details ?

File: yarn.lock [hunk 7]
```diff
@@ -460,7 +460,7 @@
     supports-color "^5.4.0"
     tslib "^1"
 
-"@oclif/command@1.5.19", "@oclif/command@^1.5.10", "@oclif/command@^1.5.13", "@oclif/command@^1.5.3", "@oclif/command@^1.5.8":
+"@oclif/command@^1.5.10", "@oclif/command@^1.5.13", "@oclif/command@^1.5.3", "@oclif/command@^1.5.8":
```
Comment (adelkahomolova) [inline yarn.lock]: remove

File: yarn.lock [hunk 8]
```diff
@@ -472,7 +472,7 @@
     debug "^4.1.1"
     semver "^5.6.0"
 
-"@oclif/config@1.13.3", "@oclif/config@^1", "@oclif/config@^1.12.6", "@oclif/config@^1.12.8":
+"@oclif/config@^1", "@oclif/config@^1.12.6", "@oclif/config@^1.12.8":
```
Comment (adelkahomolova) [inline yarn.lock]: remove

File: yarn.lock [hunk 9]
```diff
@@ -516,7 +516,7 @@
     chalk "^2.4.2"
     tslib "^1.9.3"
 
-"@oclif/plugin-help@2.2.3", "@oclif/plugin-help@^2", "@oclif/plugin-help@^2.1.6":
+"@oclif/plugin-help@^2", "@oclif/plugin-help@^2.1.6":
```
Comment (adelkahomolova) [inline yarn.lock]: remove

File: src/index.ts (hunk 6 above)
Comment (prokopsimek) [reply src/index.ts]: yes, as others commands :)

**Comment by prokopsimek** [comment]
@adelkahomolova done

File: src/index.ts [hunk 10]
```diff
@@ -14,12 +14,17 @@ class DXScannerCommand {
   static async run() {
     const cmder = new commander.Command();
 
+    //customize default help
+    cmder.name('dx-scanner').usage('[command] [options] ');
```
Comment (prokopsimek) [inline src/index.ts]: second time `name`
